x-pack/plugins/index_lifecycle_management/public/application/services/ui_metric.ts

```diff
@@ -31,49 +30,54 @@ export function init(usageCollection?: UsageCollectionSetup): void {
   }
 }
 
-export function getUiMetricsForPhases(phases: any): any {
+export function getUiMetricsForPhases(phases: Phases): any {
   const phaseUiMetrics = [
     {
       metric: UIM_CONFIG_COLD_PHASE,
-      isTracked: () => Boolean(phases[PHASE_COLD]),
+      isTracked: () => Boolean(phases.cold),
     },
     {
       metric: UIM_CONFIG_WARM_PHASE,
-      isTracked: () => Boolean(phases[PHASE_WARM]),
+      isTracked: () => Boolean(phases.warm),
     },
     {
       metric: UIM_CONFIG_SET_PRIORITY,
       isTracked: () => {
-        const phaseToDefaultIndexPriorityMap = {
-          [PHASE_HOT]: defaultHotPhase[PHASE_INDEX_PRIORITY],
-          [PHASE_WARM]: defaultWarmPhase[PHASE_INDEX_PRIORITY],
-          [PHASE_COLD]: defaultColdPhase[PHASE_INDEX_PRIORITY],
-        };
-
         // We only care about whether the user has interacted with the priority of *any* phase at all.
-        return [PHASE_HOT, PHASE_WARM, PHASE_COLD].some((phase) => {
-          // If the priority is different than the default, we'll consider it a user interaction,
-          // even if the user has set it to undefined.
-          return (
-            phases[phase] &&
-            get(phases[phase], 'actions.set_priority.priority') !==
-              phaseToDefaultIndexPriorityMap[phase]
-          );
-        });
+        const isHotPhasePriorityChanged =
+          phases.hot &&
+          phases.hot.actions.set_priority &&
+          phases.hot.actions.set_priority.priority !==
```

Comment:
Why do we have a different type for the value and the default value? Why don't we have the default value set as a number directly? (`100`)